### PR TITLE
app-dicts/myspell-en: bump to 20230101

### PR DESCRIPTION
**Pull request by kuraga**
English dictionaries for myspell/hunspell 20230101 is out:
https://extensions.libreoffice.org/extensions/english-dictionaries

Closes: https://bugs.gentoo.org/889416

Update:

* increased copyright year,
* addressed [bug #795174](https://bugs.gentoo.org/795174) (thanks to Viorel Munteanu @ceamac),
* applied #28926.

Closes: https://bugs.gentoo.org/889416
Bug: https://bugs.gentoo.org/795174
GitHub: https://github.com/gentoo/gentoo/pull/28926
Reviewed-by: Viorel Munteanu <ceamac@gentoo.org>

### CODE REVIEW

**Comment by ceamac** [comment]
@kuraga Do you think you can also address bug [795174](https://bugs.gentoo.org/795174)?  Also, if you applied #28926 here, please close it.

You have a typo in your GCO ("`Signrf-off-ny`").  It's better and easier to install `pkgdev` and commit with `pkgdev commit --signoff` instead of `git commit`, it does some extra checks and adds the GCO properly.

Thank you!

**Comment by kuraga** [comment]
@ceamac ,

> Also, if you applied #28926 here, please close it.

No, #28926 mutates old ebuilds only.

> You have a typo in your GCO ("`Signrf-off-ny`"). It's better and easier to install `pkgdev` and commit with `pkgdev commit --signoff` instead of `git commit`, it does some extra checks and adds the GCO properly.

Oooh, fixed.

> Do you think you can also address bug [795174](https://bugs.gentoo.org/795174)?

Sorry, I don't understand it.
@ceamac , BTW, could you explain it, please?
Should we add some "default" (i.e. turned on, with `+` at the front) `IUSE`/`L10N`?

> Thank you!

Thank YOU!

**Comment by ceamac** [comment]
> No, #28926 mutates old ebuilds only.

Ok.  Sorry, I didn't read it.

> Sorry, I don't understand it. @ceamac , BTW, could you explain it, please? Should we add some "default" (i.e. turned on, with `+` at the front) `IUSE`/`L10N`?

Yes, something like that.  You can take a look at `app-dicts/myspell-de`, they added a default "de" language that works.  You can do the same for en.  After that you can also drop that REQUIRED_USE if you want, because it becomes redundant.

Thank you!

**Comment by kuraga** [comment]
@ceamac , have done!

Can't you merge, can you? Seems like the maintainer @oniGino is unavailable.

Also, could you, please, write to:

> P.S. That bug isn't linked into the [`app-dicts/myspell-en`'s `Bugs` tab](https://packages.gentoo.org/packages/app-dicts/myspell-en/bugs) due to [gentoo-837191](https://bugs.gentoo.org/837191), urgh...

File: app-dicts/myspell-en/myspell-en-20230101.ebuild [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+"
+SRC_URI="https://extensions.libreoffice.org/assets/downloads/41/1672590731/dict-en-20230101_lo.oxt"
+
+LICENSE="BSD MIT LGPL-3+"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~x86-solaris"
+
+PLOCALES=( "en-AU" "en-CA" "en-GB" "en-US" "en-ZA" )
+IUSE+="+l10n_en ${PLOCALES[@]/#/l10n_}"
+REQUIRED_USE="|| ( l10n_en ${IUSE[@]} )"
```
Comment (ceamac) [inline app-dicts/myspell-en/myspell-en-20230101.ebuild:22]:
```suggestion
REQUIRED_USE="|| ( l10n_en ${PLOCALES[@]/#/l10n_} )"
```
Using IUSE in REQUIRED_USE expands to something like `"|| ( l10n_en +l10n_en ... "`, which is invalid.

You can install `pkgcheck` and run `pkgcheck scan` before committing, it validates the ebuild.

**Review by ceamac — CHANGES_REQUESTED** [review]

File: app-dicts/myspell-en/myspell-en-20230101.ebuild (hunk 1 above)
Comment (kuraga) [reply app-dicts/myspell-en/myspell-en-20230101.ebuild]: fixed.

**Comment by kuraga** [comment]
Thanks!

**Comment by ceamac** [comment]
Thank you for your contribution!

**Comment by kuraga** [comment]
@ceamac ,
I'd appreciate your attention on [gentoo-795174's comment](https://bugs.gentoo.org/795174#c3).
And, maybe, [other my pull requests](https://github.com/gentoo/gentoo/pulls/kuraga), especially `app-dicts/*`?
Thanks!

**Comment by ceamac** [comment]
I started a stabilization for myspell-en-2011, that bug can be backported straight to stable if we need to because it doesn't change any installed files, it only sets a default use flag.

Your other app-dicts PRs are very new, there are a lot of older PRs in the queue :)  Please be patient

Do you have IRC?  It's easier to talk directly, please join #gentoo-proxy-maint on [Gentoo IRC](https://www.gentoo.org/get-involved/irc-channels/) if you can.

**Comment by kuraga** [comment]
@ceamac , thanks! Addressed in #28926.

**Pull request closed without merging**